src/elfpy/simulators.py

```diff
@@ -276,93 +292,48 @@ def run_simulation(self, override_dict=None):
         up to `self.num_trading_days` days.
         """
         self.start_time = time_utils.current_datetime()
-        self.block_number = 0
+        self.last_user_action_time = 0
         self.setup_simulated_entities(override_dict)
+        last_block_in_sim = False
         for day in range(0, self.config.simulator.num_trading_days):
             self.day = day
             # Vault return can vary per day, which sets the current price per share
             if self.day > 0:  # Update only after first day (first day set to init_share_price)
                 self.market.share_price += (
-                    self.vault_apy[self.day]  # current day's apy
+                    self.config.simulator.vault_apy[self.day]  # current day's apy
                     / 365  # convert annual yield to daily
                     * self.market.init_share_price  # APR, apply return to starting price (no compounding)
                     # * self.market.share_price # APY, apply return to latest price (full compounding)
                 )
             for daily_block_number in range(self.config.simulator.num_blocks_per_day):
+                if self.day == self.config.simulator.num_trading_days - 1:
+                    if daily_block_number == self.config.simulator.num_blocks_per_day - 1:
+                        last_block_in_sim = True
```

Comment:
this might be more clear if you say `last_block_in_sim = (daily_block_number == self.config.simulator.num_blocks_per_day - 1)`